ethereum/core/src/main/java/org/hyperledger/besu/ethereum/core/encoding/DepositDecoder.java

```diff
@@ -39,6 +42,22 @@ public static Deposit decode(final RLPInput rlpInput) {
     return new Deposit(publicKey, depositWithdrawalCredential, amount, signature, index);
   }
 
+  public static Deposit decodeFromLog(final Log log) {
+    Contract.EventValuesWithLog eventValues = DepositContract.staticExtractDepositEventWithLog(log);
+    byte[] rawPublicKey = (byte[]) eventValues.getNonIndexedValues().get(0).getValue();
+    byte[] rawWithdrawalCredential = (byte[]) eventValues.getNonIndexedValues().get(1).getValue();
+    byte[] rawAmount = (byte[]) eventValues.getNonIndexedValues().get(2).getValue();
+    byte[] rawSignature = (byte[]) eventValues.getNonIndexedValues().get(3).getValue();
+    byte[] rawIndex = (byte[]) eventValues.getNonIndexedValues().get(4).getValue();
+
+    return new Deposit(
+        BLSPublicKey.wrap(Bytes.wrap(rawPublicKey)),
+        Bytes32.wrap(Bytes.wrap(rawWithdrawalCredential)),
+        GWei.of(Bytes.wrap(rawAmount).reverse().toLong()),
```

Comment:
> For Withdrawals, the CL side converted amounts into big endian before sending data to EL via engine API so we were able to keep the EL as big endian.
> 
> Was the deposit amount stored in little endian because it originated from the CL? It's shame little endian has leaked into the EL here slightly, but I guess the data is already in the contract so can't really change it after the fact 🤔

Yea exactly. The deposit contract use little endian hence we need to reverse it: https://github.com/ethereum/consensus-specs/blob/dev/solidity_deposit_contract/deposit_contract.sol#L119

I have improved the readability of the code and added a comment